ckan/lib/cli.py

```diff
@@ -2152,230 +2156,355 @@ def command(self):
+            self._update_search_params(search_data_dict)
 
-        print 'Web page resource views are being created'
+        elif not self.options.no_default_filters:
 
-        resources = model.Resource.get_all_without_views(formats)
+            self._add_default_filters(search_data_dict, view_types)
 
-        user = logic.get_action('get_site_user')({'model': model, 'ignore_auth': True}, {})
-        context = {'model': model, 'session': model.Session, 'user': user['name']}
+        if not search_data_dict.get('q'):
+            search_data_dict['q'] = '*:*'
 
-        count = 0
-        for resource in resources:
-            count += 1
-            resource_view = {'title': 'Web Page View',
-                             'description': 'View of the webpage',
-                             'resource_id': resource.id,
-                             'view_type': 'webpage_view'}
+        if ('dataset_type:dataset' not in search_data_dict['fq'] and
+                'dataset_type:dataset' not in search_data_dict['fq_list']):
+            search_data_dict['fq_list'].append('dataset_type:dataset')
 
-            logic.get_action('resource_view_create')(context, resource_view)
+        query = p.toolkit.get_action('package_search')(
+            {'ignore_capacity_check': True},
+            search_data_dict)
 
-        print '%s webpage resource views created!' % count
+        return query
 
-    def create_pdf_views(self):
-        if not p.plugin_loaded('pdf_view'):
-            print 'Please enable the pdf_view plugin to make the PDF views.'
-            return
+    def create_views(self, view_plugin_types=[]):
 
-        if not p.plugin_loaded('resource_proxy'):
-            print 'Please enable the resource_proxy plugin to make the PDF views.'
-            return
+        from ckan.lib.datapreview import add_views_to_dataset_resources
 
-        print 'PDF resource views are being created'
+        log = logging.getLogger(__name__)
 
-        resources = model.Resource.get_all_without_views(['pdf'])
+        datastore_enabled = 'datastore' in config['ckan.plugins'].split()
 
-        user = logic.get_action('get_site_user')({'model': model, 'ignore_auth': True}, {})
-        context = {'model': model, 'session': model.Session, 'user': user['name']}
+        loaded_view_plugins = self._get_view_plugins(view_plugin_types,
+                                                     datastore_enabled)
 
-        count = 0
-        for resource in resources:
-            count += 1
-            resource_view = {'title': 'PDF View',
-                             'description': 'PDF view of the resource.',
-                             'resource_id': resource.id,
-                             'view_type': 'pdf'}
+        context = {'user': self.site_user['name']}
 
-            logic.get_action('resource_view_create')(context, resource_view)
+        page = 1
+        while True:
+            query = self._search_datasets(page, loaded_view_plugins)
```

Comment:
I think because this is going through solr, it fails silently for recline views (ckan/ckanext/reclineview/plugin.py:61), since `data_dict['resource']['datastore_active']` does not exist in the resource dict when it is fetched from solr.